src/QueryM.purs

```diff
@@ -467,7 +468,18 @@ getDatumByHash hash = unwrap <$> do
   mkDatumCacheRequest DcWsp.getDatumByHashCall _.getDatumByHash hash
 
 getDatumsByHashes :: Array DataHash -> QueryM (Map DataHash Datum)
-getDatumsByHashes hashes = unwrap <$> do
+getDatumsByHashes hashes =
+  Map.mapMaybe hush <$>
+    ( unwrap <$> do
+        mkDatumCacheRequest
+          DcWsp.getDatumsByHashesCall
+          _.getDatumsByHashes
+          hashes
+    )
+
+getDatumsByHashes'
+  :: Array DataHash -> QueryM (Map DataHash (Either String Datum))
+getDatumsByHashes' hashes = unwrap <$> do
```

Comment:
I did this on purpose since usually when I do things like 

```
someFunction = 
  do 
     action
```

which I prefer, I got corrected in the review to 

```
someFunction = do 
     action
```

As that's the style in the code base.